plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -317,6 +317,58 @@ public function register_checkout_field( $options ) {
 			'autocapitalize' => empty( $options['autocapitalize'] ) ? '' : $options['autocapitalize'],
 		);
 
+		// We check if attributes are valid. This is done to prevent too much nesting and also to allow field registration
+		// even if the attributes property is invalid. We can just skip it and register the field without attributes.
+		$has_attributes = false;
+		if ( ! empty( $options['attributes'] ) ) {
+
+			if ( ! is_array( $options['attributes'] ) || 0 === count( $options['attributes'] ) ) {
+				$message = sprintf( 'An invalid attributes value was supplied when registering field with id: "%s". %s', $id, 'Attributes must be a non-empty array.' );
+				_doing_it_wrong( 'woocommerce_blocks_register_checkout_field', esc_html( $message ), '8.6.0' );
+			}
+			$has_attributes = true;
+		}
+
+		if ( $has_attributes ) {
+
+			// These are formatted in camelCase because React components expect them that way.
+			$allowed_attributes = array(
+				'placeholder',
+				'maxLength',
+				'disabled',
+				'readOnly',
+				'pattern',
+				'autocomplete',
+				'autocapitalize',
+				'title',
+			);
+
+			$valid_attributes = array_filter(
+				$options['attributes'],
+				function( $_, $key ) use ( $allowed_attributes ) {
+					return in_array( $key, $allowed_attributes, true ) || strpos( $key, 'aria-' ) === 0 || strpos( $key, 'data-' ) === 0;
+				},
+				ARRAY_FILTER_USE_BOTH
+			);
+
+			// Any invalid attributes should show a doing_it_wrong warning. It shouldn't stop field registration, though.
+			if ( count( $options['attributes'] ) !== count( $valid_attributes ) ) {
+				$invalid_attributes = array_diff_key( $options['attributes'], $valid_attributes );
+				$message            = sprintf( 'Invalid attribute found when registering field with id: "%s". Attributes: %s are not allowed.', $id, implode( ', ', $invalid_attributes ) );
+				_doing_it_wrong( 'woocommerce_blocks_register_checkout_field', esc_html( $message ), '8.6.0' );
+			}
+
+			// Escape attributes to remove any malicious code.
+			$cleaned_attributes = array_map(
+				function( $value ) {
+					return esc_attr( $value );
+				},
+				$valid_attributes
+			);
+
+			$field_data['attributes'] = $cleaned_attributes;
+		}
+
```

Comment:
Nitpick, the registration function is getting too crowded too fast, maybe we should start moving stuff to sub functions later on.

Reply:
yeah, let's follow that up, I agree.

Reply:
In this PR I moved attribute registration to its own function, I will follow up on the other settings later.

When you get chance can you re-review the changes to attribute registration?